sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/PropagateEmptyRelationSuite.scala

```diff
@@ -107,7 +112,7 @@ class PropagateEmptyRelationSuite extends PlanTest {
       val query = testRelation1
         .where(left)
         .join(testRelation2.where(right), joinType = jt, condition = Some('a.attr == 'b.attr))
-      val optimized = Optimize.execute(query.analyze)
+      val optimized = Optimize.execute(query.analyze).analyze
```

Comment:
the subsequent analyze here is to fill in timezones on casts inserted by Optimize

Reply:
We should fill the time zone in the optimizer rule